src/main/java/org/threadly/concurrent/UnfairExecutor.java

```diff
@@ -0,0 +1,348 @@
+package org.threadly.concurrent;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.locks.LockSupport;
+
+import org.threadly.util.AbstractService;
+import org.threadly.util.ArgumentVerifier;
+import org.threadly.util.Clock;
+import org.threadly.util.ExceptionUtils;
+
+/**
+ * <p>A very high performance {@link SubmitterExecutor} implementation.  Though to get those 
+ * performance gains, some guarantees are reduced.  Most prominently is execution order, this 
+ * scheduler does not ensure that tasks are executed in the order they are submitted, but rather 
+ * tasks are consumed however is fastest.</p>
+ * 
+ * <p>This executor has an execution queue per thread.  That way each thread has a many producer, 
+ * one consumer safety guarantees.  Vs normal pools which have to deal with a many to many thread 
+ * safety issue.  Determining the thread queue may be based on a variety of information, some 
+ * examples are the {@link Object#hashCode()} of the provided task, the current time, or other 
+ * weak choices to try and distribute work evenly.</p>
+ * 
+ * <p>In this it is obviously possible for one long running task to block other tasks needing to 
+ * run (even if other threads are idle).  Because of that tasks should be equally sized, having 
+ * large tasks mixed in with small ones will make this more problematic.  We also recommend having 
+ * thread counts which are prime numbers for a more even hash distribution.</p>
+ * 
+ * @author jent - Mike Jensen
+ * @since 4.5.0
+ */
+public class UnfairExecutor extends AbstractSubmitterExecutor {
+  /**
+   * Generator which will determine the task stripe by using the runnables hash code and 
+   * {@link Clock#lastKnownTimeNanos()}.  This is the fastest built in option (and thus why it is 
+   * the default).  However submissions of the same task many times without the clock being updated 
+   * can result in a scheduler being unfairly burdened.
+   */
+  public static final TaskStripeGenerator TASK_STRIPE_HASH_AND_LAST_KNOWN_TIME = new TaskStripeGenerator() {
```

Comment:
So some real world use (inside jetty) shows this to not really be feasible without a very large thread pool.  In the jetty use case we end up stacking up very unevenly on threads, and it causes request times to be very inconsistent.

In contrast the `ATOMIC_INCREMENTING` below works amazing (at least it seems so far).

So the question is...just make atomic the default?  Any ideas for other strategies?  I love how fast and simple this is, but it just is not distributing fairly.